### PR TITLE
Improve fast-sync reliability & clean up messages

### PR DESCRIPTION
**Pull request by carver**
### What was wrong?

Some sync cases were crashing:
- when no peers available for sync download
- when no receipts are received in the entire batch

Some messages are not formatted correctly, or unclear

Some unnecessary cruft remained from exploratory implementations.

### How was it fixed?

Crash fixes:
- early escape from no receipts
- catch `NoEligiblePeers`, log, and pause

Improve & fix formatting on exception messages and logs

Remove cruft (like the `complete_token`, in favor of just canceling the service)

Add a new `BaseService.cancel_nowait()` that can be called from inside the service without hanging.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by carver** [comment]
Fixes the unformatted message mentioned in #1234

File: trinity/sync/common/chain.py [hunk 1]
```diff
@@ -64,12 +61,7 @@ def __init__(self,
                  db: AsyncHeaderDB,
                  peer_pool: PeerPool,
                  token: CancelToken = None) -> None:
-        self.complete_token = CancelToken('trinity.sync.common.BaseHeaderChainSyncer.SyncCompleted')
-        if token is None:
-            master_service_token = self.complete_token
-        else:
-            master_service_token = token.chain(self.complete_token)
-        super().__init__(master_service_token)
+        super().__init__(token)
```
Comment (pipermerriam) [inline trinity/sync/common/chain.py]: cc @gsalgado ^ I'm super distracted right now but pointing you at this location as a possible candidate for how #1234 is happening.

Comment (gsalgado) [reply trinity/sync/common/chain.py]: I'd came across that yesterday when trying to find out if we were wrongly chaining tokens somewhere, but this actually looked correct to me

File: trinity/sync/full/chain.py [hunk 2]
```diff
@@ -94,8 +94,16 @@ class FastChainSyncer(BaseHeaderChainSyncer):
             # TODO implement the maximum task size at each step instead of this magic number
             max_headers = min((MAX_BODIES_FETCH, MAX_RECEIPTS_FETCH)) * 4
             batch_id, headers = await self.header_queue.get(max_headers)
-            await self._process_headers(headers)
-            self.header_queue.complete(batch_id, headers)
+            try:
+                await self._process_headers(headers)
+            except NoEligiblePeers:
+                self.logger.info(
+                    f"No available peers to sync with, retrying in {self.NO_PEER_RETRY_PAUSE}s"
```
[the remark this answers is not in the document]
Comment (carver) [reply trinity/sync/full/chain.py]: Yup, starting in Py3.6. So we can't use it in py-evm yet, only p2p and trinity.

**Comment by carver** [comment]
Tests are green, Circle is just waiting for the SSH session to time out. Merging...

**Pull request merged**